src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTest.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection.Metadata;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.EditAndContinue.UnitTests
+{
+    internal sealed partial class EditAndContinueTest : IDisposable
+    {
+        private readonly CSharpCompilationOptions? _options;
+        private readonly TargetFramework _targetFramework;
+
+        private readonly List<IDisposable> _disposables = new();
+        private readonly List<GenerationInfo> _generations = new();
+
+        private bool _hasVerified;
+
+        public EditAndContinueTest(CSharpCompilationOptions? options, TargetFramework? targetFramework)
+        {
+            _options = options;
+            _targetFramework = targetFramework ?? TargetFramework.Standard;
+        }
+
+        internal EditAndContinueTest AddGeneration(string source, Action<GenerationVerifier> verification)
```

Comment:
Maybe we should name this parameter 'validator' across the board a la CompileAndVerify

https://github.com/dotnet/roslyn/blob/0c92c3f34bb6ea1e5f04d544670573a711beda52/src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs#L816-L828

Reply:
Good idea.. I'm making this change right now in my browser. Hopefully committing it works!